sdk/src/main/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayData.java

```diff
@@ -0,0 +1,508 @@
+
+    public String getNamespace() {
+      return ns;
+    }
+
+    public String getKey() {
+      return key;
+    }
+    /**
+     * Retrieve the {@link DisplayData.Type} of display metadata. All metadata conforms to a
+     * predefined set of allowed types.
+     */
+    public Type getType() {
+      return type;
+    }
+
+    /**
+     * Retrieve the value of the metadata item.
+     */
+    public String getValue() {
+      return value;
+    }
+
+    /**
+     * Return the optional short value for an item. Types may provide a short-value to displayed
+     * instead of or in addition to the full {@link Item#value}.
+     */
+    public String getShortValue() {
+      return shortValue;
+    }
+
+    /**
+     * Retrieve the optional label for an item. The label is a human-readable description of what
+     * the metadata represents. UIs may choose to display the label instead of the item key.
+     *
+     * <p>If no label was specified, this will return {@code null}.
+     */
+    public String getLabel() {
+      return label;
+    }
+
+    /**
+     * Retrieve the optional link URL for an item. The URL points to an address where the reader
+     * can find additional context for the display metadata.
+     *
+     * <p>If no URL was specified, this will return {@code null}.
+     */
+    public String getUrl() {
+      return url;
+    }
+
+    @Override
+    public String toString() {
+      return getValue();
+    }
+
+    private Item withLabel(String label) {
+      return new Item(this.ns, this.key, this.type, this.value, this.shortValue, this.url, label);
+    }
+
+    private Item withUrl(String url) {
+      return new Item(this.ns, this.key, this.type, this.value, this.shortValue, url, this.label);
+    }
+  }
+
+  /**
+   * Unique identifier for a display metadata item within a component.
+   * Identifiers are composed of the key they are registered with and a namespace generated from
+   * the class of the component which registered the item.
+   *
+   * <p>Display metadata registered with the same key from different components will have different
+   * namespaces and thus will both be represented in the composed {@link DisplayData}. If a
+   * single component registers multiple metadata items with the same key, only the most recent
+   * item will be retained; previous versions are discarded.
+   */
+  public static class Identifier {
+    private final String ns;
+    private final String key;
+
+    static Identifier of(Class<?> namespace, String key) {
+      return new Identifier(namespace.getName(), key);
+    }
+
+    private Identifier(String ns, String key) {
+      this.ns = ns;
+      this.key = key;
+    }
+
+    public String getNamespace() {
+      return ns;
+    }
+
+    public String getKey() {
+      return key;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (obj instanceof Identifier) {
+        Identifier that = (Identifier) obj;
+        return this.ns.equals(that.ns)
```

Comment:
```
return Objects.equals(this.ns, that.ns)
    && Objects.equals(this.key, that.key);
```

Preferable for symmetry, and because it handles null values better.